examples/js/loaders/deprecated/LegacyGLTFLoader.js

```diff
@@ -1616,26 +1662,74 @@ THREE.LegacyGLTFLoader = ( function () {
 									geometry.addAttribute( 'uv', bufferAttribute );
 									break;
 
-								case 'TEXCOORD_1':
-									geometry.addAttribute( 'uv2', bufferAttribute );
-									break;
-
 								case 'COLOR_0':
 								case 'COLOR0':
 								case 'COLOR':
 									geometry.addAttribute( 'color', bufferAttribute );
 									break;
 
+								case 'WEIGHT_0':
+								case 'WEIGHT0':
 								case 'WEIGHT':
 									geometry.addAttribute( 'skinWeight', bufferAttribute );
 									break;
 
+								case 'JOINT_0':
+								case 'JOINT0':
 								case 'JOINT':
 									geometry.addAttribute( 'skinIndex', bufferAttribute );
 									break;
 
+								default:
+
+									var addAttributeWithItsNameUnchanged = false;
+
+									var underscoreIndex = attributeId.indexOf( '_' );
+									if ( underscoreIndex !== - 1 ) {
+
+										var attrName = attributeId.slice( 0, underscoreIndex ).toLowerCase();
+										var attrNum = parseInt( attributeId.slice( underscoreIndex + 1 ) ) + 1;
+
+										switch ( attrName ) {
+
+											case 'position':
+											case 'normal':
+											case 'color':
+												geometry.addAttribute( attrName + attrNum, bufferAttribute );
+												break;
+
+											case 'texcoord':
+												geometry.addAttribute( 'uv' + attrNum, bufferAttribute );
+												break;
+
+											case 'weight':
+												geometry.addAttribute( 'skinWeight' + attrNum, bufferAttribute );
+												break;
+
+											case 'joint':
+												geometry.addAttribute( 'skinIndex' + attrNum, bufferAttribute );
+												break;
+
+											default:
+												addAttributeWithItsNameUnchanged = true;
+										}
+
+									}
+
+									if ( addAttributeWithItsNameUnchanged || underscoreIndex === - 1 ) {
+
+										var material = json.materials[ primitive.material ];
+										var attributeNames = json.techniques[ material.technique ].attributes;
+										var attributeName = Object.keys( attributeNames )[ attributeIndex ];
+
+										geometry.addAttribute( attributeName, bufferAttribute );
+
+									}
+
```

Comment:
If we decide not to rename attributes three.js does not use itself, I believe this `default:` block can be written more simply:

```js
geometry.addAttribute( attributeId, bufferAttribute );
```

I imagine you're renaming attributes for consistency with the existing code, but it isn't strictly necessary for attributes whose names don't have inherent meaning in three.js, and I'd be more comfortable merging this PR if we can avoid adding two new nested `switch` statements to accomplish it.

Reply:
attributeId represents constants like 'POSITION' and 'TEXCOORD_0', but the first argument to geometry.addAttribute() needs to be the name of the corresponding variable in the shader text. That's why I need to do a couple handstands to find the name of the variable in the GLTF file through the material's technique. But I can simplify the code to do only that, which will be a maybe five-line change total.

Reply:
Oh I see, thanks!